### PR TITLE
Improve error handling in migrated controllers

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR aims making error handling the same across all migrated controllers.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Part of https://github.com/PrestaShop/PrestaShop/issues/12438
| How to test?  | Nothing should change for migrated pages "Profiles", "Contacts" and "Currencies"

### Controllers that error handling were updated:
* ProfileController
* CurrencyController

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12505)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php [hunk 1]
```diff
@@ -181,24 +189,23 @@ public function bulkDeleteAction(Request $request)
     /**
      * Get human readable error for exception.
      *
-     * @param ProfileException $e
-     *
-     * @return string Error message
+     * @return array
      */
-    protected function handleDeleteException(ProfileException $e)
+    protected function getErrorMessages()
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php:203]: it's a BC: sadly you have to deprecate the old one and create a new one (why do we need this function to be protected?)

File: src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig [hunk 2]
```diff
@@ -25,6 +25,16 @@
 
 {% extends '@PrestaShop/Admin/layout.html.twig' %}
 
+{% set layoutTitle = 'Profiles'|trans({}, 'Admin.Navigation.Menu') %}
+{% set enableSidebar = true %}
+{% set layoutHeaderToolbarBtn = {
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig]: not really fan of moving business logic into the templates!

Let's keep this in the controllers please :+1:

**Review by mickaelandrieu — DISMISSED** [review]
I like the refactoring, but I disagree on moving some business logic into the templates.

Revert this change and it's :+1:  for me.

Good job @sarjon

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php (hunk 1 above)
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php:203]: it's not a BC, this controller should be part of 1.7.6. For `protected` method, i don't know, maybe to make it extendable if developers decide to extend this controller, either way, i can make it `private` if you want. :)

File: src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig (hunk 2 above)
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig]: displaying button is not a business logic and in my opinion it belongs in template, no? why add these lines of code in controller and pass it directly to template?

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig]: the display stays in the template, the data in the controller: separation of concerns 👍

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php (hunk 1 above)
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php:203]: do you mean that ``handleDeleteException`` function doesn't exist in 1.7.5.x branch?

Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php:203]: whole `ProfilesController` does not exist in 1.7.5.x branch

File: src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig (hunk 2 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Profiles/index.html.twig]: Agreed :+1:

**Comment by sarjon** [comment]
@mickaelandrieu @PierreRambaud data was moved to controller. :+1:

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thanks @sarjon !

**Pull request merged**